src/app/core/components/admin/admin.interceptor.ts

```diff
@@ -0,0 +1,68 @@
+import {HttpErrorResponse, HttpEvent, HttpHandler, HttpInterceptor, HttpRequest} from '@angular/common/http';
+import {BehaviorSubject, Observable, throwError} from 'rxjs';
+import {Injectable} from '@angular/core';
+import {catchError, filter, switchMap, take} from 'rxjs/operators';
+import {Router} from '@angular/router';
+
+import {AdminAuthService} from './services/admin-auth.service';
+
+@Injectable()
+export class AdminInterceptor implements HttpInterceptor {
+
+  private isRefreshing = false;
+  private refreshTokenSubject: BehaviorSubject<any> = new BehaviorSubject<any>(null);
+
+  constructor(public authService: AdminAuthService, private router: Router) {
+  }
+
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+
+    if (this.authService.getAccessToken()) {
+      request = this.addToken(request, this.authService.getAccessToken());
+    }
+
+    return next.handle(request).pipe(catchError((error: HttpErrorResponse) => {
+      if (error instanceof HttpErrorResponse && error.status === 401) {
+        return this.handle401Error(request, next);
+      } else {
```

Comment:
You don't need else if you return something from if statement

Reply:
You are right